### PR TITLE
Adds a few extra exosuit modules (Take 2)

### PR DESCRIPTION
**Pull request by githubuser4141**
## About The Pull Request

Exosuits here sadly don't have a lot of tools compared to other servers, so I finally got around to porting some exo-tools from Aurora.

## Why It's Good For The Game

It's fun, the additional tools add variety to mech gameplay!
Also: Tweaks exosuit repair so that quick-repairs are no longer possible, because exosuits shouldn't be able to be band'aid'ed back together like a human.
Adds 2 exosuit GLs that fire either frag or stingball projectiles, should be useful for police exosuits. Adds the exosuit cannon, a 8-shot gun that fire low-speed, hi-damage rounds with moderate AP & adds the exosuit RPG, a 4-shot gun that fires nerfed HE rocket rounds.

Increases the max_damage & integrity of most hardsuits as a lot of mechsuits didn't feel very durable after limb-specific targeting was added by SPCR, and since an arm is actually *both* arms 75 HP for an arms is 75 HP before your mech is *completely useless*. E.g., a 600 HP mechsuit chassis looks intimidating but less so when taking out it's arms/legs mission kills it.

Increases mecha armour's energy resists by a fair amount because, holy heck the lightfall was doing 30+ damage to mechs through combat armor ⭕ 

## Changelog
:cl:
add: Adds the exosuit stingball/frag launcher, exosuit RPG launcher & exosuit cannon. [Sprites from Aurora!]
Adds gun SFX from Polaris.
tweak: Sorts the exosuit components into their component files rather then in premade. [Idea-Port from Bay.]
Adds colorful text to exosuit descs [Credit to Bay] & adds a Dwarf Fortress reference
balance: You can't quickpatch exosuits anymore. To repair an exosuit you have to dismantle it and weld/wire the individual components.
Buffs the integrity of exosuit components. Exosuit tracks can no longer strafe. Increaes the energy resist on standard, ablative and combat mech armour.
/:cl:

Thanks to TheShown1911 for code advice.

### CODE REVIEW

**Comment by githubuser4141** [comment]
If anyone could say why the checks are fucked it'd be appreciated.

File: code/modules/mechs/components/_components.dm [hunk 1]
```diff
@@ -56,7 +56,9 @@
 
 /obj/item/mech_component/proc/update_health()
 	total_damage = brute_damage + burn_damage
-	if(total_damage > max_damage) total_damage = max_damage
+//	if(total_damage > max_damage) //total_damage = max_damage
+	if(total_damage < (max_damage - max_damage)) // Bandaid fix for mechs overhealing
```
Comment (Pink-Chink) [inline code/modules/mechs/components/_components.dm:60]:
```suggestion
	if(total_damage < 0) // Bandaid fix for mechs overhealing
```

File: code/modules/mechs/components/_components.dm [hunk 2]
```diff
@@ -67,23 +69,33 @@
 /obj/item/mech_component/proc/ready_to_install()
 	return TRUE
 
+
 /obj/item/mech_component/proc/repair_brute_damage(amt)
 	take_brute_damage(-amt)
+	if(total_damage <= 0)
+		return
```
Comment (Pink-Chink) [inline code/modules/mechs/components/_components.dm:76]:
```suggestion
	if(total_damage <= 0)
		return
	take_brute_damage(-amt)
```
Is this what you meant to do?

File: code/modules/mechs/components/_components.dm [hunk 3]
```diff
@@ -67,23 +69,33 @@
...
 
 /obj/item/mech_component/proc/repair_burn_damage(amt)
 	take_burn_damage(-amt)
+	if(total_damage <= 0)
+		return
```
Comment (Pink-Chink) [inline code/modules/mechs/components/_components.dm:81]:
```suggestion
	if(total_damage <= 0)
		return
	take_burn_damage(-amt)
```

File: code/modules/mechs/components/_components.dm [hunk 4]
```diff
@@ -56,7 +56,9 @@
 
 /obj/item/mech_component/proc/update_health()
 	total_damage = brute_damage + burn_damage
-	if(total_damage > max_damage) total_damage = max_damage
+//	if(total_damage > max_damage) //total_damage = max_damage
+	if(total_damage < 0) // Bandaid fix for mechs overhealing
+		total_damage = 0
```
Comment (Pink-Chink) [inline code/modules/mechs/components/_components.dm:61]: This was previously `if(total_damage < (max_damage - max_damage))`. I fat fingered the commit button with my suggestion.

File: code/modules/mechs/components/_components.dm [hunk 5]
```diff
@@ -67,23 +69,33 @@
...
+		return
+
 
 /obj/item/mech_component/proc/take_brute_damage(amt)
 	brute_damage += amt
 	update_health()
 	if(total_damage == max_damage)
 		take_component_damage(amt,0)
+	if(amt < 1)
+		return
```
Comment (Pink-Chink) [inline code/modules/mechs/components/_components.dm:90]:
```suggestion
	if(amt < 1)
		return
	if(total_damage == max_damage)
		take_component_damage(amt,0)
```

File: code/modules/mechs/components/arms.dm [hunk 6]
```diff
@@ -44,3 +44,45 @@
...
+	action_delay = 10
+	max_damage = 125
+	power_use = 50
+	matter = list(MATERIAL_STEEL = 15, MATERIAL_PLASTEEL = 5, MATERIAL_PLASMA = 4, MATERIAL_DIAMOND = 2)
+
+/obj/item/mech_component/manipulators/heavy
+	name = "heavy arms"
+	exosuit_desc_string = "super-heavy reinforced manipulators"
+	icon_state = "heavy_arms"
+	desc = "Designed for durability and dishing out beatings, this heavy set of manipulators can both take and dish out beatings."
```
Comment (DiaFRAME444) [inline code/modules/mechs/components/arms.dm:83]: Uhh, how about "Designed for prolonged combat situations, this heavy set of manipulators can both take and dish out beatings."

**Comment by githubuser4141** [comment]
Done! Should be ready 4 review now.

File: code/modules/mechs/components/_components.dm (hunk 5 above)
Comment (githubuser4141) [reply code/modules/mechs/components/_components.dm]: Please review, I changed the code to something else =)

**Comment by Pink-Chink** [comment]
Good timing. I'll include this in the test server.

**Comment by githubuser4141** [comment]
> Good timing. I'll include this in the test server.

Cool, I swapped the HE rocket to a flash round with a rocket sprite due to a horrific bug related to explosions and floors which can crash the server. (Happened when firing a stock RPG.) And uh.. fixed the mech desc upon examine to not repeat itself.

File: code/modules/mechs/equipment/combat.dm [hunk 7]
```diff
@@ -7,99 +7,290 @@
...
+/obj/item/gun/energy/flamer/mounted
+	bad_type = /obj/item/gun/energy/flamer/mounted
+	spawn_tags = null
+
+/////
+
+/obj/item/mech_equipment/mounted_system/launcher
+	name = "launcher thing"
+	desc = "You shouldn't be seeing this."
+	icon_state = "mech_missile_pod"
```
Comment (Pink-Chink) [inline code/modules/mechs/equipment/combat.dm:198]:
```suggestion
	icon_state = "mech_missile_pod"
	bad_type = /obj/item/mech_equipment/mounted_system/launcher
```

File: code/modules/mechs/equipment/combat.dm [hunk 8]
```diff
@@ -7,99 +7,290 @@
 		return C.charge/C.maxcharge
 	return null
 
-/obj/item/mech_equipment/mounted_system/taser
+/obj/item/mech_equipment/mounted_system/energy
+	name = "energy thing"
+	desc = "You shouldn't be seeing this."
+	icon_state = "mecha_taser"
```
Comment (Pink-Chink) [inline code/modules/mechs/equipment/combat.dm:13]:
```suggestion
	icon_state = "mecha_taser"
	bad_type = /obj/item/mech_equipment/mounted_system/energy
```

**Comment by DiaFRAME444** [comment]
> Increases mecha armour's energy resists by a fair amount because, holy heck the lightfall was doing 30+ damage to mechs through combat armor

Ehhh, not a fan of nerfing the underused damage type because of one gun. I'm not even sure how that works, Lightfall should deal 14 damage to something with 30 armor.

**Comment by githubuser4141** [comment]
> 

...? It's energy /rifles/ in general. Also, no, the LF's has a 1.3x dam mult, which means it does 39 damage - and a 1.2 divisor - so with the combat armor's original 11 rating? It'd do about 30 damage, which is significant for a relatively common rifle that: has a scope, is hitscan & is relatively efficient.

Remember that the toughest mech leg or arm, has around 280 HP - so that's less then 10 shots to mission kill a mech with a scoped, hitscan weapon that's not an EMP or rocket. Increasing the combat armour's energy resist to 16 marginally less - around 26 damage so that's a few extra shots to kill it. Ablative armor doesn't matter because: It's specialized against lasers, and absolutely useless against bullets.

Other energy rifles, such as the cannon & Valkyire will still absolutely wreck an exosuit, esp. the Valk which ablative won't protect against.

File: code/modules/mechs/equipment/combat.dm (hunk 7 above)
Comment (githubuser4141) [reply code/modules/mechs/equipment/combat.dm:199]: That's probably better.

File: code/modules/mechs/components/arms.dm [hunk 9]
```diff
@@ -44,3 +44,45 @@
...
+	exosuit_desc_string = "industrial lifter arms"
+	icon_state = "loader_arms"
+	melee_damage = 15
+	max_damage = 90
+	power_use = 30
+
+/obj/item/mech_component/manipulators/light
+	name = "light arms"
+	exosuit_desc_string = "lightweight, segmented manipulators"
+	desc = "As flexible as they are fragile, these Vey-Med manipulators can follow a pilot's movements in close to real time."
```
Comment (Pink-Chink) [inline code/modules/mechs/components/arms.dm:60]:
```suggestion
	desc = "As flexible as they are fragile, these manipulators can follow a pilot's movements in close to real time."
```
Vey-Med isn't a thing here.

**Review by Firefox13 — COMMENTED** [review]
Both grenade launchers (frag version should cost more resources because it's lethal, by the way) and missile rack seem too strong, mostly because they can shit out seemingly unending mayhem.
I dunno what to do to missile rack short of commenting it out, but grenade launchers are probably fine. Lower their magazines to six because muh revolving magazines.

File: code/modules/mechs/premade/misc.dm [hunk 10]
```diff
@@ -31,11 +29,33 @@
...
 	mech_health = 350 //Default is 300, so 50 more HP then the power loader. Worse then the combat chassis as it requires sensors.
-
 	power_use = 5
-	climb_time = 30 // Awkward to get in/out of as it's intended for spacepod use
-	matter = list(MATERIAL_STEEL = 25, MATERIAL_PLASTEEL = 5, MATERIAL_GLASS = 10)
+	climb_time = 30
+	matter = list(MATERIAL_STEEL = 25, MATERIAL_PLASTEEL = 7, MATERIAL_GLASS = 12)
 	has_hardpoints = list(HARDPOINT_BACK)
-	desc = "A rugged design originally intended for light EVA crafts, this chassis has been refitted for exosuit usage. It's surprisingly durable for its cost."
+	desc = "A rugged design originally intended for S, this chassis has been refitted for exosuit usage. It's surprisingly durable for its cost."
```
Comment (Firefox13) [inline code/modules/mechs/premade/misc.dm:38]: S?

**Comment by Firefox13** [comment]
Also, those sprites. Yuck.

**Comment by githubuser4141** [comment]
> Both grenade launchers (frag version should cost more resources because it's lethal, by the way) and missile rack seem too strong, mostly because they can shit out seemingly unending mayhem. I dunno what to do to missile rack short of commenting it out, but grenade launchers are probably fine. Lower their magazines to six because muh revolving magazines.

Would nerfing the missiles be enough? It's currently a 30 dam / 90 halloss projectile right now. Compared to the PKM which fires 5 rounds of PK ammo in a quick burst and has infinite ammo or the taser which which fires 3 hitscan lasers it's actually quite underwhelming. I mean, I'll nerf it to like 60 halloss I guess.

**Comment by thedarkcrusader** [comment]
Maybe instead of nerfing base mis

> Both grenade launchers (frag version should cost more resources because it's lethal, by the way) and missile rack seem too strong, mostly because they can shit out seemingly unending mayhem. I dunno what to do to missile rack short of commenting it out, but grenade launchers are probably fine. Lower their magazines to six because muh revolving magazines.

perhaps it be best to make the missle rack take actual ammo like the PKm etc etc and i would personally use the weaker rpg rocket we have makeshift rpg rocket  used for the artist revolver made from the crafting station

**Comment by githubuser4141** [comment]
this is silly, going to re-open in a more compact format.

**Pull request closed without merging**